src/components/Visualizer.vue

```diff
@@ -103,23 +79,32 @@ export default {
         this.visualizer = this.visualizer == this.visualizerArr.length - 1 ? 0 : this.visualizer + 1;
     },
     update() {
-        if(!this.autoUpdate) return;
+        if(!this.autoUpdate || !this.ctx) return;
```

Comment:
Are you taking `ctx` as a constructor argument? If so, can `this.ctx` ever be `null` or falsey?

Reply:
I have added a `beforeDestory` event to set the ctx to null in order to stop the update event. It might be redundant, need more testing.

Reply:
See line 107 in this file

Reply:
In other words, this `beforeDestroy` lifecycle event is to free any resources associated with the currently live `AudioContext` right?

Reply:
Ahh, that makes sense. That's where I was confused about, because while I was debugging this, the update function are still running returning undefined errors. Thus I have added this, turns out it was because of a `setTimeout` is running. I have then fixed the bug but still kept this here just in case. Now this should be removed without issues.